### ISSUE TITLE
LMMS displays templates as last opened project

### ISSUE THREAD

**Issue by Sti2nd** [LMMS/lmms#1812]
Since I have edited the default template, and that is loaded on startup, LMMS ALWAYS says that default.mpt was the last project. So it should not say it for default.mpt, at least

![image](https://cloud.githubusercontent.com/assets/6373430/6448629/9c89b8e0-c11b-11e4-8d21-3c35055061b5.png)

**Comment by Sti2nd** [comment]
PS: I like how the slashes are mixed between forward and backward

**Comment by tresf** [comment]
> PS: I like how the slashes are mixed between forward and backward

Yeah that is a cleanup task for us to tackle on a rainy day. :)

> LMMS [...] says that `default.mpt` was the last project. 

Good find.  We should remove all templates from that list via [`ConfigManager.cpp#L192`](https://github.com/LMMS/lmms/blob/master/src/core/ConfigManager.cpp#L192).


Something like this perhaps... 
```cpp
void ConfigManager::addRecentlyOpenedProject( const QString & _file )
{
	////  ADD TEMPLATE EXTENSION CHECK
	if ( ! _file.endsWith( ".mpt" ) ) {
		m_recentlyOpenedProjects.removeAll( _file );
		if( m_recentlyOpenedProjects.size() > 15 )
		{
			m_recentlyOpenedProjects.removeLast();
		}
		m_recentlyOpenedProjects.push_front( _file );
		ConfigManager::inst()->saveConfigFile();
	}
}
```

**Comment by M374LX** [comment]
Actually, the template used is shown as the last opened project every time one creates a project from a template, not only from the default one. This should not happen with any template at all.

A better solution than @tresf's suggestion is to check the file name within `Song::loadProject()`, which is called from `Song::createNewProjectFromTemplate()`.

At `Song.cpp` line 1045, we could replace:
```C++
ConfigManager::inst()->addRecentlyOpenedProject( fileName );
```
with
```C++
if ( !fileName.endsWith( ".mpt" ) )
{
    ConfigManager::inst()->addRecentlyOpenedProject( fileName );
}
```

**Comment by tresf** [comment]
Yes, much cleaner!

**Comment by M374LX** [comment]
One more detail is that creating a new project from a template created with an old version of LMMS show a message like
`This project was created with LMMS version x.x.x, but version x.x.x is installed`
It is better to use the word "template" instead of "project".

**Comment by tresf** [comment]
> One more detail is that creating a new project from a template created with an old version of LMMS show a message like
This project was created with LMMS version x.x.x, but version x.x.x is installed
It is better to use the word "template" instead of "project".

We should probably file a separate issue for that.

**Comment by tresf** [comment]
@M374LX actually, I think my proposed solution is more scalable.  Why put the logic in an unrelated calss?

**Comment by M374LX** [comment]
@tresf 
I have just seen that `ConfigManager::addRecentlyOpenedProject()` is also called from `Song::guiSaveProject()`. So, in case you prefer a saved project saved templates not to be listed as recent projects, your solution is really better.

**Comment by tresf** [comment]
:+1:

**Comment by M374LX** [comment]
> We should probably file a separate issue for that.

Filed: #1869

**Comment by tresf** [comment]
Closed via: #2091.  Added to 1.2.0 changelog.

-Tres
